### PR TITLE
test: Correct JSON-RPC serialization tests and add edge case coverage

### PR DESCRIPTION
**Pull request by pankaj-bind**
# Description
This update addresses a failing test in `tests/server/apps/jsonrpc/test_serialization.py` and significantly enhances the test coverage for the JSON-RPC serialization layer to ensure robustness against various edge cases.

The `test_handle_unicode_characters` test was previously failing due to an invalid JSON-RPC request payload. This has been corrected to use a valid `message/send` request, allowing the test to properly validate Unicode handling.

Additionally, new tests have been added to cover scenarios that were previously untested:
-   **Malformed JSON**: Ensures the server correctly identifies and rejects invalid JSON structures, returning a `JSONParseError`.
-   **Oversized Payloads**: Checks how the application handles requests that exceed size limits, now gracefully handled as an `InvalidRequestError`.
-   **Unicode Characters**: The corrected test now properly validates that the application can handle Unicode characters in request payloads without serialization issues.

A minor improvement was also made to `src/a2a/server/apps/jsonrpc/jsonrpc_app.py` to explicitly catch `HTTPException` (like the one for oversized payloads) and convert it into a standard JSON-RPC error response, ensuring consistent error reporting to clients.

These changes ensure more robust and reliable JSON-RPC communication, which is critical for agentic applications.

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Follow the [`CONTRIBUTING` Guide](https://github.com/a2aproject/a2a-python/blob/main/CONTRIBUTING.md).
- [ ] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
  - Important Prefixes for [release-please](https://github.com/googleapis/release-please):
    - `fix:` which represents bug fixes, and correlates to a [SemVer](https://semver.org/) patch.
    - `feat:` represents a new feature, and correlates to a SemVer minor.
    - `feat!:`, or `fix!:`, `refactor!:`, etc., which represent a breaking change (indicated by the `!`) and will result in a SemVer major.
- [ ] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/a2a/server/apps/jsonrpc/jsonrpc_app.py [hunk 1]
```diff
@@ -232,6 +233,13 @@ async def _handle_requests(self, request: Request) -> Response:
                 request_id,
                 A2AError(root=InvalidRequestError(data=json.loads(e.json()))),
             )
+        except HTTPException as e:
+            if e.status_code == 413: # Payload Too Large
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply src/a2a/server/apps/jsonrpc/jsonrpc_app.py]: If this status code works, I'd recommend using it.

**Review by holtskinner — APPROVED** [review]
